src/main/scala/scalaz/stream/async/mutable/Queue.scala

```diff
@@ -45,10 +45,24 @@ trait Queue[A] {
    *
    * Please use `Topic` instead of `Queue` when all subscribers
    * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
    */
   def dequeue: Process[Task, A]
 
   /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
```

Comment:
Hmm, I really think throwing an IllegalArgumentException eagerly seems like the right thing to do. The stack trace will point to the immediate caller, even if the caller's caller is lazy, and this might be useful information. Delaying throwing until later is going to result in the stack trace being useless, and silently treating negative values as valid seems more likely to paper over what is probably a user bug. And you can always do:

```scala
(if (limit <= 0) q.dequeueAvailable else q.dequeueBatch(limit))
```

Not sure how `njoin` and `boundedQueue` slipped through the cracks, but I would fix those as well! :)